packages/nuxt3/src/pages/runtime/composables.ts

```diff
@@ -8,3 +9,29 @@ export const useRouter = () => {
 export const useRoute = () => {
   return useNuxtApp()._route as RouteLocationNormalizedLoaded
 }
+
+export interface PageMeta {
+  [key: string]: any
+  transition?: false | TransitionProps
+  layout?: false | string | Ref<false | string> | ComputedRef<false | string>
+  // TODO: https://github.com/vuejs/vue-next/issues/3652
+  // keepalive?: false | KeepAliveProps
+}
+
+declare module 'vue-router' {
+  interface RouteMeta extends PageMeta {}
+}
+
+const warnRuntimeUsage = (method: string) =>
+  console.warn(
+    `${method}() is a compiler-hint helper that is only usable inside ` +
+      '<script setup> of a single file component. Its arguments should be ' +
+      'compiled away and passing it at runtime has no effect.'
+  )
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const definePageMeta = (meta: PageMeta): void => {
+  if (!process.dev) {
```

Comment:
I guess you meant `if (process.dev)` to eliminate from prod bundle?

```suggestion
  if (process.dev) {
```

Reply:
This is following the pattern of Vue's macros - warning only in production.